internal/controller/csiaddons/persistentvolumeclaim_controller.go

```diff
@@ -346,8 +400,26 @@ func (r *PersistentVolumeClaimReconciler) SetupWithManager(mgr ctrl.Manager, ctr
 		},
 	}
 
+	scPred := predicate.Funcs{
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			if e.ObjectNew == nil || e.ObjectOld == nil {
+				return false
```

Comment:
what happens when the case where someone deletes the annotation on the storageclass? have you tested that case?

Reply:
deleting annotation will just Reconcile PVCs which doesn't have annotations set.

Reply:
but it should reconcile all the pvc's which belong to this SC, i means the controller added annotation need to be removed when a annotation is removed on the sc as well